src/fairchem/core/_cli_hydra.py

```diff
@@ -0,0 +1,126 @@
+"""
+Copyright (c) Facebook, Inc. and its affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+from __future__ import annotations
+
+import logging
+import os
+from typing import TYPE_CHECKING
+
+import hydra
+
+if TYPE_CHECKING:
+    import argparse
+
+    from omegaconf import DictConfig
+
+from submitit import AutoExecutor
+from submitit.helpers import Checkpointable, DelayedSubmission
+from torch.distributed.launcher.api import LaunchConfig, elastic_launch
+
+from fairchem.core.common import distutils
+from fairchem.core.common.flags import flags
+from fairchem.core.common.utils import get_timestamp_uid, setup_env_vars, setup_imports
+from fairchem.core.components.runner import Runner
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class Submitit(Checkpointable):
+    def __call__(self, dict_config: DictConfig, cli_args: argparse.Namespace) -> None:
+        self.config = dict_config
+        self.cli_args = cli_args
+        # TODO: setup_imports is not needed if we stop instantiating models with Registry.
+        setup_imports()
+        setup_env_vars()
+        try:
+            distutils.setup(map_cli_args_to_dist_config(cli_args))
+            runner: Runner = hydra.utils.instantiate(dict_config.runner)
+            runner.load_state()
+            runner.run()
+        finally:
+            distutils.cleanup()
+
+    def checkpoint(self, *args, **kwargs):
+        logging.info("Submitit checkpointing callback is triggered")
+        new_runner = Runner()
+        new_runner.save_state()
+        logging.info("Submitit checkpointing callback is completed")
+        return DelayedSubmission(new_runner, self.config)
+
+
+def map_cli_args_to_dist_config(cli_args: argparse.Namespace) -> dict:
+    return {
+        "world_size": cli_args.num_nodes * cli_args.num_gpus,
+        "distributed_backend": "gloo" if cli_args.cpu else "nccl",
+        "submit": cli_args.submit,
+        "summit": None,
+        "cpu": cli_args.cpu,
+        "use_cuda_visibile_devices": True,
+    }
+
+
+def get_hydra_config_from_yaml(
+    config_yml: str, overrides_args: list[str]
+) -> DictConfig:
+    # Load the configuration from the file
+    os.environ["HYDRA_FULL_ERROR"] = "1"
```

Comment:
Is this read somewhere?

Reply:
is used by hydra to split out full error messages